### ISSUE TITLE
multiple default homeservers

### ISSUE THREAD

**Issue by hex-m** [vector-im/element-web#17135]
#### Is your suggestion related to a problem? Please describe.

Our organization has a setup with multiple Synapse instances, because we have/want users with different domain parts in their Matrix-ID.

To make the login-flow as easy as possible we have a separate Element-Web instance for every Synapse instance. This has drawbacks:
* Users are confused that we have multiple chat services.
* Sending the URL of a room to someone can lead to login errors because they'd have to use a different Element-Web instance.

#### Describe the solution you'd like.
The possibility to specify multiple servers in `default_server_configs` and give users an easy way to choose from them.

#### Describe alternatives you've considered.
An alternative solution would be support for multiple user-domains in Synapse. https://github.com/matrix-org/synapse/issues/8885

**Comment by t3chguy** [comment]
A single synapse for auth can be used to act as a frontend for multiple Matrix servers, during `/login` the server sends back the `server_name` to which to connect to post-auth. 
Equally, you could just enter the user's MXID into the login screen and the server would be automatically resolved from that.

**Comment by hex-m** [comment]
> you could just enter the user's MXID into the login screen

That is true that the correct homeserver would be discovered when you enter an MXID, but our users don't know what an MXID is when they first login. Also: we use SSO (SAML) which means (as I understand it) that the homeserver has to be fixed already when you forward the users to the IDP.

> A single synapse for auth [..]

Not sure if I grasp this. Is there documentation for how to set this up?

**Comment by t3chguy** [comment]
I don't know if there's documentation for that but there are definitely some deployments doing that.
Does your SSO vary depending on which server or is it all the same and depending on the user going through the SSO the target matrix server should be different?

**Comment by hex-m** [comment]
The IDP is always the same. The authorization decision is currently based on `attribute_requirements`. So yes, the target homeserver should be chosen depending on some attribute of the user.

**Comment by t3chguy** [comment]
Pretty sure there is a deployment functioning exactly like that and the `/login` request returns back the right matrix base_url to use for that user. Probably best reaching out to the synapse support room to see if they know more about it.
